postgres-appliance/scripts/upload_pg_log_to_s3.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import boto
+import math
+import os
+import logging
+import subprocess
+import sys
+import time
+
+from datetime import datetime, timedelta
+from filechunkio import FileChunkIO
+
+logger = logging.getLogger(__name__)
+
+
+def compress_pg_log():
+    yesterday = datetime.now() - timedelta(days=1)
+    yesterday_day_number = yesterday.strftime('%u')
+
+    log_file = os.path.join(os.getenv('PGLOG'), 'postgresql-' + yesterday_day_number + '.csv')
+    archived_log_file = os.path.join(os.getenv('LOG_TMPDIR'), yesterday.strftime('%F') + '.csv.gz')
+
+    if os.path.getsize(log_file) == 0:
+        logger.warning("Postgres log from yesterday '%s' is empty.", log_file)
+        sys.exit(0)
+
+    with open(archived_log_file, 'wb') as f_out:
+        subprocess.Popen(['gzip', '-9c', log_file], stdout=f_out).wait()
```

Comment:
I'd catch all exceptions there and write them nicely to the log before aborting. More often than not we see some unreadable stack traces in the logs (i.e. zmon), it would be better to format those errors for readability.